src/functions-and-arrays.js

```diff
@@ -1,16 +1,51 @@
 // Iteration #1: Find the maximum
+const maxOfTwoNumbers = (num1, num2) => (num1 > num2 ? num1 : num2);
+
+maxOfTwoNumbers(2, 4);
 
 // Iteration #2: Find longest word
 const words = ['mystery', 'brother', 'aviator', 'crocodile', 'pearl', 'orchard', 'crackpot'];
 
+function findLongestWord(words) {
+let longest = '';
+for (let i = 0; i < words.length; i++) {
+if (words[i].length > longest.length)
+longest = words[i];
+}
+return longest;
+}
+
+console.log(findLongestWord(words));
+
+
 // Iteration #3: Calculate the sum
 
 const numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumNumbers(numbers){
+    let total=0;
+    for(let i in numbers) { 
+        total += numbers[i];
+    }
+    return total;
+}
+sumNumbers(numbers);
+
+
 // Iteration #4: Calculate the average
 // Level 1: Array of numbers
 const numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+
+function averageNumbers(numbersAvg){
+    let total=0;
+    for(let i in numbersAvg) { 
+        total += numbersAvg[i]/numbersAvg.length;
+    }
+    return total;
+}
+averageNumbers(numbersAvg);
```

Comment:
The same implementation for the empty array is expected here. The concept is the same as mentioned in the comment above.